File: python-stack/functions/read.py
from nitric.api.exception import NotFoundException, NitricServiceException
from nitric.faas import start, HttpContext
from nitric.api import Documents
import json


async def handler(ctx: HttpContext) -> HttpContext:
    try:
        doc_id = ctx.req.path.split("/")[-1]
    except IndexError:
        ctx.res.body = "Invalid request"
        ctx.res.status = 400
        return ctx

    try:
        example = await Documents().collection("examples").doc(doc_id).get()
        ctx.res.body = json.dumps(example.content)
    except NotFoundException:
        ctx.res.status = 404
        ctx.res.body = "Example not found"
    except NitricServiceException:
        ctx.res.status = 500
        ctx.res.body = "An unexpected error occurred"

    return ctx


if __name__ == "__main__":
    start(handler)

File: react-site-stack/services/functions/read.ts
import { faas, documents } from "@nitric/sdk";
import { Example, path } from "../common";

// Start your function here
faas
  .http(async (ctx: faas.HttpContext): Promise<faas.HttpContext> => {
    // get params from path
    const { id } = path.test(ctx.req.path);

    if (!id) {
      ctx.res.body = new TextEncoder().encode("Invalid Request");
      ctx.res.status = 400;
    }

    try {
      console.log("getting doc id", id);
      const example = await documents()
        .collection<Example>("examples")
        .doc(id)
        .get();

      ctx.res.json(example);
    } catch (e) {
      ctx.res.status = 404;
      ctx.res.body = new TextEncoder().encode(
        `Example not found!: ${e.message}`
      );
    }

    return ctx;
  })
  .start();
